src/vcpkg-utils.ts

```diff
@@ -177,7 +169,24 @@ export class Utils {
   }
 
   public static getAllCachedPaths(baselib: baselib.BaseLib, vcpkgRootDir: string): string[] {
-    let paths = runvcpkglib.getOrdinaryCachedPaths(vcpkgRootDir);
+    const runnerOS = process.env.ImageOS ? process.env.ImageOS : process.platform;
+    const isWindows = runnerOS.slice(0,3) == "win";
+
+    let directorySeparator = '/';
+    if (isWindows) {
+      directorySeparator = '\\';
+    }
+
+    let paths = [ vcpkgRootDir,
+                  "!" + vcpkgRootDir + directorySeparator + "packages",
+                  "!" + vcpkgRootDir + directorySeparator + "buildtrees",
+                  "!" + vcpkgRootDir + directorySeparator + "downloads" ];
+
+    if (isWindows) {
+        paths.push(process.env.LOCALAPPDATA + "\\vcpkg");
+    } else {
+        paths.push(process.env.HOME + "/.cache/vcpkg");
+    }
```

Comment:
I tried caching both `VPCKG_ROOT/packages` and `CMAKE_BINARY_DIR/vcpkg_installed` but neither worked. Only caching these locations outside of both the vcpkg root and the CMake build directory works.

Reply:
if i look this right, the variable `VCPKG_DEFAULT_BINARY_CACHE` is not being set by the action.
My suggestion is that this action sets the env var value in order to explicitly drive where the `vcpkg` is going to install the built libraries.

Reply:
Overriding that variable does not seem right. I think the most robust way to handle this would be checking if `VCPKG_DEFAULT_BINARY_CACHE` is set, and if cache that location, otherwise use the default. https://vcpkg.io/en/docs/users/binarycaching.html#configuration

Reply:
There are many default locations that vcpkg can write the binary caching to, so the `run-vcpkg` action cannot know 100% what was the place `vcpkg` put the binary cache. This is the reason that the action must drive and be certain of the location.

I agree that as you stated, if `VCPKG_DEFAULT_BINARY_CACHE` is set and it is valid (assuming _valid_. means existing and writeable, something the action must check), the action must not override it. On the other hand if not set, the action should set it to a sensible and valid value.